File: packages/create-sst/bin/presets/examples/rest-api-python/templates/backend/functions/get.py
import json
from db.notes import getNotes

def main(event, context):
  notes = getNotes()
  noteId = event["pathParameters"]["id"]

  if noteId in notes:
    return {
      "statusCode": 200,
      "body": json.dumps(notes[noteId], indent=2)
    }

  return {
    "statusCode": 404,
    "body": json.dumps({
      "error": True
    })
  }

File: examples/crud-api-dynamodb/services/functions/get.ts
import AWS from "aws-sdk";
import { Table } from "@serverless-stack/node/table";

const dynamoDb = new AWS.DynamoDB.DocumentClient();

import { APIGatewayProxyHandlerV2 } from "aws-lambda";

export const main: APIGatewayProxyHandlerV2 = async (event) => {
  const params = {
    // Get the table name from the environment variable
    TableName: Table.Notes.tableName,
    // Get the row where the noteId is the one in the path
    Key: {
      userId: "123",
      noteId: event.pathParameters.id,
    },
  };
  const results = await dynamoDb.get(params).promise();

  return results.Item
    ? {
        statusCode: 200,
        body: JSON.stringify(results.Item),
      }
    : {
        statusCode: 404,
        body: JSON.stringify({ error: true }),
      };
};
